File: Problems/4. Median of Two Sorted Arrays/Median of Two Sorted Arrays.js
var findMedianSortedArrays = function (nums1, nums2) {
    const arr = [...nums1, ...nums2].sort((a, b) => a - b);

    if (arr.length % 2 === 0) {
        return (arr[arr.length / 2 - 1] + arr[arr.length / 2]) / 2;
    }

    return arr[Math.floor(arr.length / 2)];
};

console.log(findMedianSortedArrays([3], [-2, -1]));
console.log(findMedianSortedArrays([2, 4, 6], [1, 3, 5]));

File: Problems/4. Median of Two Sorted Arrays/Median of Two Sorted Arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        sorted_arrays = sorted(nums1 + nums2)
        mid = len(sorted_arrays) >> 1
        if not len(sorted_arrays) & 1:
            return (sorted_arrays[mid] + sorted_arrays[mid - 1]) / 2
        else:
            return sorted_arrays[mid]


# print(Solution.findMedianSortedArrays(0, [1, 3], [2, 4]))
# print(Solution.findMedianSortedArrays(0, [1, 3], [2]))
# print(Solution.findMedianSortedArrays(0, [2, 4, 6], [1, 3, 5]))
# print(Solution.findMedianSortedArrays(0, [3], [-2, -1]))
print(Solution.findMedianSortedArrays(0, [1, 2, 5], [3, 6, 7]))
